app/libs/finders/user_typable_internship_offer.rb

```diff
@@ -36,10 +36,11 @@ def nearby_query_part(query, coordinates)
                                longitude: coordinates.longitude)
     end
 
+    # school_members gather students and other profiles/roles
     def school_members_query
       query = InternshipOffer.all
 
-      if user.try(:middle_school?) && school_type_param != 'high_school'
+      if user.try(:class_room).try(:troisieme_generale?)
```

Comment:
```suggestion
      if user.try(:class_room).try(:professional_school_track?)
```
je pense ca se repercute donc aussi ici